### PR TITLE
 convert project to gradle Multi-module

### PR DESCRIPTION
**Pull request by PittyXu**

### CODE REVIEW

**Comment by kdhrubo** [comment]
@PittyXu - its a great effort. However it would be good to stick to maven.

**Comment by kdhrubo** [comment]
I will take a stab at converting this into maven multi-module.

**Comment by kdhrubo** [comment]
@PittyXu looks like you have done the mavenized multimodule starter now?

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+plugins {
+    `java-library`
+    `maven-publish`
+    id("org.springframework.boot") version "3.2.4"
```
Comment (thadguidry) [inline app/build.gradle.kts:4]: bump Spring Boot to 3.2.5 throughout

**Comment by kdhrubo** [comment]
Also it will be good to remove gradle on favor of maven to minimize
confusion

**Comment by thadguidry** [comment]
@kdhrubo ah, yes, we should have both builders.  So you prefer Maven and want to stick with that and have them remove all the Gradle stuff, correct?

**Comment by kdhrubo** [comment]
Yes only maven..out pipelines all depend on maven

We will now release the starters as well

One more major issue with this is missing oracle oracle 9i module

So effectively we need one additional.moduele for.oracle.9i and also another app pulling in that starter.

**Comment by PittyXu** [comment]
@kdhrubo I delete the gradle code and convert project to Multi-module project with minimal modifications. Subsequent structured adjustments may be necessary, such as adjusting test cases to db2rest module, configrations to starters.

**Comment by kdhrubo** [comment]
I will look into your branch and check as much as I can. 
I suspect there will be few bumps after merge so lets try to minimize as far as possible.
However this is a very necessary change.

**Comment by kdhrubo** [comment]
We need to do some internal refactoring before we can move on this. Mongo and JDBC has common dependencies needs to be separated. 
I will keep this open. Will start on the internal refactoring.

**Comment by kdhrubo** [comment]
@PittyXu refactoring to module or libs is pretty much down now. 
Hence closing this pull request. 

Now in next releases we will come back to work on starters. 
This clears the path for db2grpc i.e grpc support

Let us know what you think, by joining the discord channel. Appreciate your feedback.

**Pull request closed without merging**